agent/transformers/opentsdb.go

```diff
@@ -0,0 +1,100 @@
+package transformers
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/sensu/sensu-go/types"
+)
+
+// OpenTSDBList contains a list of OpenTSDB metrics
+type OpenTSDBList []OpenTSDB
+
+// OpenTSDB contains values of an OpenTSDB metric
+type OpenTSDB struct {
+	Name      string
+	Value     float64
+	TagSet    []*types.MetricTag
+	Timestamp int64
+}
+
+// Transform transforms metrics in OpenTSDB format to Sensu Metric Format
+func (o OpenTSDBList) Transform() []*types.MetricPoint {
+	var points []*types.MetricPoint
+	for _, metric := range o {
+		mp := &types.MetricPoint{
+			Name:      metric.Name,
+			Value:     metric.Value,
+			Timestamp: metric.Timestamp,
+			Tags:      metric.TagSet,
+		}
+		points = append(points, mp)
+	}
+	return points
+}
+
+// ParseOpenTSDB parses OpenTSDB metrics into a list of OpenTSDB structs
+func ParseOpenTSDB(output string) (OpenTSDBList, error) {
+	openTSDBList := OpenTSDBList{}
+
+	// Split each line of the output into its own metric
+	metrics := strings.Split(output, "\n")
+
+	for _, metric := range metrics {
+		parts := strings.Split(metric, " ")
+
+		// Ensure we have all the required components. A single metric requires a
+		// name, timestamp, value and at least one tag.
+		if len(parts) < 4 {
+			return nil, fmt.Errorf("invalid opentsdb metric: %s", metric)
```

Comment:
It might be helpful to have a more descriptive error message here, pertaining to minimum # of fields required of an opentsdb metric.

Reply:
Done!